ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -412,7 +417,11 @@ void read(DataStreamRequestByteBuf request, ChannelHandlerContext ctx,
           }
           buf.release();
           return null;
-        }, executor));
+        }, executor)).exceptionally(exception -> {
+      replyDataStreamException(server, exception, info.getRequest(), request, ctx);
+      buf.release();
+      return null;
+    });
```

Comment:
Use whenComplete(..) and also a try-catch as below:
```suggestion
        }, executor)
    ).whenComplete((v, exception) -> {
      try {
        if (exception != null) {
          replyDataStreamException(server, exception, info.getRequest(), request, ctx);
        }
      } finally {
        buf.release();
      }
    });
```